crates/matrix-sdk-base/src/room/tombstone.rs

```diff
@@ -12,18 +12,291 @@
+    ///
+    /// [`m.room.tombstone`]: https://spec.matrix.org/v1.14/client-server-api/#mroomtombstone
+    /// [`m.room.create`]: https://spec.matrix.org/v1.14/client-server-api/#mroomcreate
+    pub fn predecessor_room(&self) -> Option<PredecessorRoom> {
+        self.create_content().and_then(|content_event| content_event.predecessor).map(
+            |predecessor| PredecessorRoom {
+                room_id: predecessor.room_id,
+                event_id: predecessor.event_id,
+            },
+        )
+    }
+}
+
+/// When a room A is tombstoned, it is replaced by a room B. The room A is the
+/// predecessor of B, and B is the successor of A. This type holds information
+/// about the successor room. See [`Room::successor_room`].
+///
+/// A room is tombstoned if it has received a [`m.room.tombstone`] state event.
+///
+/// [`m.room.tombstone`]: https://spec.matrix.org/v1.14/client-server-api/#mroomtombstone
+#[derive(Debug)]
+pub struct SuccessorRoom {
+    /// The ID of the next room replacing this (tombstoned) room.
+    pub room_id: OwnedRoomId,
+
+    /// The reason why the room has been tombstoned.
+    pub reason: Option<String>,
+}
+
+/// When a room A is tombstoned, it is replaced by a room B. The room A is the
+/// predecessor of B, and B is the successor of A. This type holds information
+/// about the predecessor room. See [`Room::predecessor_room`].
+///
+/// To know the predecessor of a room, the [`m.room.create`] state event must
+/// have been received.
+///
+/// [`m.room.tombstone`]: https://spec.matrix.org/v1.14/client-server-api/#mroomtombstone
+/// [`m.room.create`]: https://spec.matrix.org/v1.14/client-server-api/#mroomcreate
+#[derive(Debug)]
+pub struct PredecessorRoom {
+    /// The ID of the old room.
+    pub room_id: OwnedRoomId,
+
+    /// The event ID of the last known event in the predecesssor room.
+    pub event_id: OwnedEventId,
+}
+
+#[cfg(test)]
+mod tests {
+    use std::ops::Not;
+
+    use assert_matches::assert_matches;
+    use matrix_sdk_test::{
+        async_test, event_factory::EventFactory, JoinedRoomBuilder, StateTestEvent,
+        SyncResponseBuilder,
+    };
+    use ruma::{event_id, room_id, user_id};
+    use serde_json::json;
+
+    use crate::{
+        store::{RoomLoadSettings, StoreConfig},
+        BaseClient, RoomState, SessionMeta,
+    };
+
+    async fn base_client() -> BaseClient {
+        let base_client =
+            BaseClient::new(StoreConfig::new("cross-process-store-locks-name".to_owned()));
+
+        base_client
+            .activate(
+                SessionMeta {
+                    user_id: user_id!("@alice:example.org").into(),
+                    device_id: ruma::device_id!("AYEAYEAYE").into(),
+                },
+                RoomLoadSettings::default(),
+                #[cfg(feature = "e2e-encryption")]
+                None,
+            )
+            .await
+            .unwrap();
+
+        base_client
+    }
```

Comment:
No chance this helper could be reused? Looks like if it's needed here, it would be needed in every other file that makes use of a `BaseClient` for testing purposes…

Reply:
Of course, we have `logged_in_base_client` already. Updated.